querybuilder/rule.go

```diff
@@ -72,41 +90,55 @@ func (r *Rule) getInputValue(dataset map[string]interface{}) interface{} {
 		} else if rresult.Kind() == reflect.Slice && i != (steps-1) {
 			result = rresult.Index(0)
 		}
-
-		if result == nil {
-			return nil
-		}
 	}
 
-	iv := r.parseValue(result)
+	iv, err := r.parseValue(result)
+
 	if r.Sanitize && r.Type == "string" {
 		v := iv.(string)
-		return sanitize(&v)
+		return sanitize(&v), err
 	}
 
-	return iv
+	return iv, err
 }
 
-func (r *Rule) parseValue(v interface{}) interface{} {
+func (r *Rule) parseValue(v interface{}) (interface{}, error) {
 	rv := reflect.ValueOf(v)
 
+	var errg errgroup.Group
+	var wg sync.WaitGroup
+
 	if rv.Kind() == reflect.Slice {
 		sv := make([]interface{}, rv.Len())
 
-		for i, vv := range v.([]interface{}) {
-			sv[i] = r.castValue(vv)
+		wg.Add(rv.Len())
+		for _, vv := range v.([]interface{}) {
+			go func(vv interface{}) {
+				errg.Go(func() error {
+					var err error
+					_, err = r.castValue(vv)
+					if err != nil {
+						return err
+					}
+					return nil
+				})
+			}(vv)
+			wg.Done()
```

Comment:
if we are running `castValue()` on a bunch of inputs and say `castValue()` returns a non-nil `err`, do you think we should proceed to other inputs at all?